src/python/pants/backend/python/goals/setup_py.py

```diff
@@ -589,15 +586,47 @@ async def generate_chroot(request: SetupPyChrootRequest) -> SetupPyChroot:
                 f"`entry_point='{entry_point.module}:main'. See {url}."
             )
         entry_point_requests.append(ResolvePexEntryPointRequest(binary[PexEntryPointField]))
+
     binary_entry_points = await MultiGet(
         Get(ResolvedPexEntryPoint, ResolvePexEntryPointRequest, request)
         for request in entry_point_requests
     )
+    source_with_binaries = dict()
     for key, binary_entry_point in zip(key_to_binary_spec.keys(), binary_entry_points):
-        entry_points = setup_kwargs.setdefault("entry_points", {})
-        console_scripts = entry_points.setdefault("console_scripts", [])
         if binary_entry_point.val is not None:
-            console_scripts.append(f"{key}={binary_entry_point.val.spec}")
+            source_with_binaries[key] = binary_entry_point.val.spec
+
+    # Collect entry points from `python_distribution(entry_points=...)`
+    source_entry_points = dict()
+    if exported_target.target.has_field(PythonDistributionEntryPoints):
+        entry_points_field = await Get(
+            ResolvedPythonDistributionEntryPoints,
+            ResolvePythonDistributionEntryPointsRequest(
+                exported_target.target[PythonDistributionEntryPoints]
+            ),
+        )
+
+        if entry_points_field.val:
+            for key, section in entry_points_field.val.items():
+                source_entry_points[key] = {
+                    name: entry_point.spec for name, entry_point in section.items()
+                }
+
+    # Collect any entry points from the setup_py() object.
```

Comment:
One of the things I was struggling with at first is seeing how all the pieces fit together. This might help:

```suggestion
    # Collect any entry points from the setup_py() object. Note that this was 
    # already normalized in `python_artifact.py`.
```